dbms/src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -98,11 +98,11 @@ BlockIO InterpreterDropQuery::execute()
 		}
 		else
 		{
-			/// Удаляем метаданные и саму таблицу из оперативки.
+			/// Delete table data
+			table.first->drop();
+			/// Delete table metdata and table itself from memory
 			database->removeTable(current_table_name);
 
-			/// Удаляем данные таблицы
-			table.first->drop();		/// TODO Не удалять метаданные, если таблицу не получилось удалить.
```

Comment:
Call to `database->removeTable` removes metadata, but table could throw exception in `drop` method and refuse to be removed. In that case we get dangling table (it will not be loaded after server restart, but its files will be in place and could be unexpectedly reused after next CREATE).